examples/platform/silabs/LEDWidget.cpp

```diff
@@ -82,3 +81,57 @@ void LEDWidget::Animate()
         }
     }
 }
+uint8_t LEDWidget::GetLED()
+{
+    return mLed;
+}
+bool LEDWidget::GetLEDStatus(uint8_t led)
+{
+    return mLedStatus;
+}
+#if (defined(SL_MATTER_RGB_LED_ENABLED) && SL_MATTER_RGB_LED_ENABLED == 1)
+void RGBLEDWidget::SetLevel(uint8_t level)
+{
+    mLevel = level;
+}
+uint8_t RGBLEDWidget::GetLevel()
+{
+    return mLevel;
+}
+void RGBLEDWidget::SetColor(uint8_t red, uint8_t green, uint8_t blue)
+{
+    if (GetLEDStatus(GetLED()))
+    {
+        ChipLogProgress(Zcl, "SetColor : %u|%u|%u", red, green, blue);
+        GetPlatform().SetLedColor(GetLED(), red, green, blue);
+    }
+}
+void RGBLEDWidget::GetColor(uint16_t r, uint16_t g, uint16_t b)
```

Comment:
This will not get anything r,g,b will be local copies. You need to use references or pointers.